front/src/modules/trainschedule/components/ManageTrainSchedule/UpdateTrainSchedulesV2.tsx

```diff
@@ -0,0 +1,112 @@
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+import { useStoreDataForRollingStockSelector } from 'modules/rollingStock/components/RollingStockSelector/useStoreDataForRollingStockSelector';
+import checkCurrentConfig from 'modules/trainschedule/components/ManageTrainSchedule/helpers/checkCurrentConfig';
+import { setFailure, setSuccess } from 'reducers/main';
+import type { OsrdConfState } from 'reducers/osrdconf/types';
+import { updateSelectedTrainId } from 'reducers/osrdsimulation/actions';
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+
+import formatTrainSchedulePayload from './helpers/formatTrainSchedulePayload';
+
+type UpdateTrainSchedulesProps = {
+  setIsWorking: (isWorking: boolean) => void;
+  setDisplayTrainScheduleManagement: (arg0: string) => void;
+  setTrainResultsToFetch: (trainScheduleIds?: number[]) => void;
+  trainIdToEdit: number | undefined;
+  setTrainIdToEdit: (trainScheduleIds?: number) => void;
+};
+
+// Refacto in a component to prepare the migration of the patch in editoast (need to be a React component to use hooks inside like rtk's)
+
+const UpdateTrainSchedulesV2 = ({
+  setIsWorking,
+  setDisplayTrainScheduleManagement,
+  setTrainResultsToFetch,
+  trainIdToEdit,
+  setTrainIdToEdit,
+}: UpdateTrainSchedulesProps) => {
+  const { getConf, getName, getDepartureTime, getStartTime } = useOsrdConfSelectors();
+  const confName = useSelector(getName);
+  const simulationConf = useSelector(getConf);
+  const departureTime = useSelector(getDepartureTime);
+  const startTime = useSelector(getStartTime);
+
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+
+  const [putV2TrainScheduleById] = osrdEditoastApi.endpoints.putV2TrainScheduleById.useMutation();
+
+  const { rollingStock } = useStoreDataForRollingStockSelector();
+
+  async function submitConfUpdateTrainSchedules() {
+    const formattedSimulationConf = checkCurrentConfig(
+      simulationConf as OsrdConfState,
+      t,
+      dispatch,
+      rollingStock?.name
+    );
+
+    if (formattedSimulationConf && trainIdToEdit) {
+      setIsWorking(true);
+      let callSuccess = true;
+      const trainSchedule = formatTrainSchedulePayload(
+        formattedSimulationConf,
+        confName,
+        startTime
+      );
+      try {
+        await putV2TrainScheduleById({
+          id: trainIdToEdit,
+          trainScheduleForm: trainSchedule,
+        })
+          .unwrap()
+          .catch((e) => {
+            callSuccess = false;
+            dispatch(setFailure(castErrorToFailure(e)));
+          });
+
+        if (callSuccess) {
+          setTrainResultsToFetch([trainIdToEdit]);
+          dispatch(
+            setSuccess({
+              title: t('trainUpdated'),
+              text: `${confName}: ${departureTime}`,
```

Comment:
Use the new util created in the `TrainSettings.tsx` comment

```suggestion
              text: `${confName}: ${convertIsoUtcToLocalTime(startTimeFromStore).substring(0, 19)}`,
```